e2e/test/scenarios/permissions/sandboxes.cy.spec.js

```diff
@@ -505,257 +505,242 @@ describeEE("formatting > sandboxes", () => {
-         *  - In a real-world scenario it is quite possible for an admin to save that SQL question without running it first. This fails!
-         *  (more info: https://github.com/metabase/metabase-enterprise/issues/520#issuecomment-772528159)
-         *
-         * That's why this test has 2 versions that reflect both scenarios. We'll call them "normal" and "workaround".
-         * Until the underlying issue is fixed, "normal" scenario will be skipped.
-         *
-         * Related issues: metabase#10474, metabase#14629
-         */
-
-        // skipping the workaround test because the function `runAndSaveQuestion`
-        // relies on the existence of a save button on a saved question that is not dirty
-        // which is a bug fixed in issue metabase#14302
-        ["normal" /* , "workaround" */].forEach(test => {
-          it(
-            `${test.toUpperCase()} version:\n advanced sandboxing should not ignore data model features like object detail of FK (metabase-enterprise#520)`,
-            { tags: "@quarantine" },
-            () => {
-              cy.intercept("POST", "/api/card/*/query").as("cardQuery");
-              cy.intercept("PUT", "/api/card/*").as("questionUpdate");
-
-              cy.createNativeQuestion({
-                name: "EE_520_Q1",
-                native: {
-                  query:
-                    "SELECT * FROM ORDERS WHERE USER_ID={{sandbox}} AND TOTAL > 10",
-                  "template-tags": {
-                    sandbox: {
-                      "display-name": "Sandbox",
-                      id: "1115dc4f-6b9d-812e-7f72-b87ab885c88a",
-                      name: "sandbox",
-                      type: "number",
-                    },
+        cy.findByTestId("TableInteractive-root")
+          .findByText("Awesome Concrete Shoes")
+          .click();
+        popover()
+          .findByText(/View details/i)
+          .click();
+
+        cy.log(
+          "It should show object details instead of filtering by this Product ID",
+        );
+        cy.findByTestId("object-detail");
+        cy.findAllByText("McClure-Lockman");
+      });
+
+      /**
+       * This issue (metabase-enterprise#520) has a peculiar quirk:
+       *  - It works ONLY if SQL question is first run (`result_metadata` builds), and then the question is saved.
+       *  - In a real-world scenario it is quite possible for an admin to save that SQL question without running it first. This fails!
+       *  (more info: https://github.com/metabase/metabase-enterprise/issues/520#issuecomment-772528159)
+       *
+       * That's why this test has 2 versions that reflect both scenarios. We'll call them "normal" and "workaround".
+       * Until the underlying issue is fixed, "normal" scenario will be skipped.
+       *
+       * Related issues: metabase#10474, metabase#14629
+       */
+      ["normal", "workaround"].forEach(test => {
```

Comment:
The "workaround" test passes now after changing `runAndSaveQuestion` into `runQuestion`.
It appears it's not flaky, see stress test: https://github.com/metabase/metabase/actions/runs/11723164294/job/32654242720

I still kept it in `@quarantine` for now. I created an issue to deal with this: #49671